pex/dependency_configuration.py

```diff
@@ -152,9 +212,17 @@ def register(parser):
         type=str,
         action="append",
         help=(
-            "Specifies a transitive requirement to override when resolving. Any distribution "
-            "requirement in the PEX's resolve that matches the override project name is replaced "
-            "with the given override requirement. This option can be used multiple times."
+            "Specifies a transitive requirement to override when resolving. Overrides can either "
+            "modify an existing dependency on a project name by changing extras, version "
+            "constraints or markers or else they can completely swap out the dependency for a "
+            "dependency on another project altogether. For the former, simply supply the "
+            "requirement you wish. For example, specifying `--override cowsay==5.0` will override "
+            "any transitive dependency on cowsay that has any combination of extras, version "
+            "constraints or markers with the requirement `cowsay==5.0`. To completely replace "
+            "cowsay with another library altogether, you can specify an override like "
+            "`--override cowsay=my-cowsay>2`. This will replace any transitive dependency on "
+            "cowsay that has any combination of extras, version constraints or markers with the "
+            "requirement `my-cowsay>2`. This option can be used multiple times."
```

Comment:
@cburroughs hopefully its now more clear exactly what `--override` can be expected to do. I do not go into gory detail about the ability to say complex things like:
```console
--override 'a=b; os_name == "posix"' \
--override 'a=c; os_name != "posix" and python_version >= "3"' \
--override 'a=d; os_name != "posix" and python_version < "3"'
```
But as noted elsewhere, this works and is tested; so I'd like to just leave it as one of those use cases that you might expect should work and then turns out it does.

Please verify the `--help` is now a bit more clear (or not).

Reply:
@cburroughs I'm going to get a release out tonight but I'll circle back to address any feedback you have.